spec/amp-html-layout.md

```diff
@@ -0,0 +1,159 @@
+
+### `layout`
+
+The optional layout attribute allows specifying how the component behaves in the document layout.
+Valid values for the layout attribute are:
+
+- Not present: The `layout` will be inferred as following:
+  - if `width` equals to `auto` `fixed-height` layout is assumed;
+  - if `width` or `height` attributes are present `fixed` layout is assumed;
+  - if `width` and `height` are not present `container` layout is assumed
+- `fixed`: The `width` and `height` attributes must be present. The only exceptions are `amp-pixel`
+and `amp-audio` elements.
+- `fixed-height`: The `height` attribute must be present. The `width` attribute must not be present
+or must be equal to `auto`.
+- `responsive`: The `width` and `height` attributes must be present and are used to determine the
+aspect ratio of the component and the component is sized to the width of its container element while
+maintaining the height based on the aspect ratio.
+- `fill`: Element size will be determined by the parent element.
+- `container`: The component is assumed to not have specific layout itself but only act as a
+container. Its children as rendered immediately.
+- `nodisplay`: The component takes up zero space on the screen as if its display style was `none`.
+The `width` and `height` attributes are not required.
+
+Each element documents which `layout` values it supported. If an element does not support the
+specified value it would trigger a runtime error.
+
+### `media`
+
+All AMP custom elements support the `media` attribute. The value of media is a media query. If the query does not match, the element is not rendered at all and it's resources and potentially it's child resources will not be fetched. If the browser window changes size or orientation the media queries are re-evaluated and elements are hidden and shown based on the new results.
+
+Example: Here we have 2 images with mutually exclusive media queries. Depending on the screen width one or the other will be fetched and rendered. Note that the media attribute is available on all custom elements, so it can be used with non-image elements such as ads.
+
+```html
+    <amp-img
+        media="(min-width: 650px)"
+        src="wide.jpg"
+        width=466
+        height=355 layout="responsive" ></amp-img>
+    <amp-img
+        media="(max-width: 649px)"
+        src="narrow.jpg"
+        width=527
+        height=193 layout="responsive" ></amp-img>
+```
+
+### `placeholder`
+
+The `placeholder` attribute can be set on any HTML element, not just AMP elements. It indicates that
+the element marked with this attribute acts as a placeholder for the parent AMP element. If specified
+a placeholder element must be a direct child of the AMP element. By default, the placeholder is
+immediately shown for the AMP element, even if the AMP element's resources have not been downloaded
+or initialized. Once ready the AMP element typically hides placeholder and shows its own content.
```

Comment:
Consider 'hides (the / it's) placeholder' as an alternative to 'hides placeholder'.